service/src/main/java/bio/terra/workspace/service/workspace/CloudSyncRoleMapping.java

```diff
@@ -110,23 +116,90 @@ public class CloudSyncRoleMapping {
               "cloudbuild.builds.approve")
           .build();
 
-  private static final CustomGcpIamRole PROJECT_READER =
-      CustomGcpIamRole.of("PROJECT_READER", PROJECT_READER_PERMISSIONS);
-  private static final CustomGcpIamRole PROJECT_WRITER =
-      CustomGcpIamRole.of("PROJECT_WRITER", PROJECT_WRITER_PERMISSIONS);
-  private static final CustomGcpIamRole PROJECT_OWNER =
-      CustomGcpIamRole.of("PROJECT_OWNER", PROJECT_OWNER_PERMISSIONS);
-  // Currently, workspace editors, applications and owners have the same cloud permissions as
-  // writers. If that changes, create a new CustomGcpIamRole and modify the map below.
-  public static final ImmutableMap<WsmIamRole, CustomGcpIamRole> CUSTOM_GCP_PROJECT_IAM_ROLES =
-      ImmutableMap.of(
-          WsmIamRole.OWNER, PROJECT_OWNER,
-          // TODO: this should map to PROJECT_APPLICATION if that's created.
-          WsmIamRole.APPLICATION, PROJECT_WRITER,
-          WsmIamRole.WRITER, PROJECT_WRITER,
-          WsmIamRole.READER, PROJECT_READER);
+  private final List<String> projectReaderDataprocPermissions =
+      new ImmutableList.Builder<String>()
+          .addAll(projectReaderPermissions)
+          .add(
+              "dataproc.clusters.get",
+              "dataproc.clusters.list",
+              "dataproc.jobs.get",
+              "dataproc.jobs.list")
+          .build();
+
+  private final List<String> projectOwnerDataprocPermissions =
+      new ImmutableList.Builder<String>()
+          .addAll(projectReaderDataprocPermissions)
+          .addAll(projectOwnerPermissions)
+          .add(
+              // Dataproc CRUD permissions
+              // TODO: Move read permissions to projectReader and revoke modify permissions once
```

Comment:
please attach ticket to TODO.
what does revoke modify permissions once mean?

Reply:
Attached ticket and updated comment. Remove all permissions in projectOwnerDataprocPermissions role and then run cloudIamSync endpoint.